Specs/Core/loadWithXhrSpec.js

```diff
@@ -133,18 +143,269 @@ defineSuite([
             });
         });
 
-        xit('can support 2xx HTTP status (other than 200)', function(){
+        xit('can support 2xx HTTP status (other than 200)', function() {
             return loadWithXhr({
-                method: 'POST',
-                url: 'http://jsonplaceholder.typicode.com/posts',
-                data: {
-                    title: 'foo',
-                    body: 'bar',
-                    userId: 1
+                method : 'POST',
+                url : 'http://jsonplaceholder.typicode.com/posts',
+                data : {
+                    title : 'foo',
+                    body : 'bar',
+                    userId : 1
                 }
-            }).then(function(result){
+            }).then(function(result) {
                 expect(JSON.parse(result).id).toEqual(101);
             });
         });
     });
+
+    describe('URL loading using XHR', function() {
+        describe('returns a promise that rejects when the request', function() {
+            it('results in an HTTP status code greater than or equal to 300', function() {
+                return loadWithXhr({
+                    url : 'http://example.invalid'
+                }).then(function(resolvedValue) {
+                    expect(resolvedValue).toBeUndefined();
+                }, function(rejectedError) {
+                    expect(rejectedError instanceof RequestErrorEvent).toBe(true);
+                });
+            });
+
+            it('loads an invalid JSON string response with a json responseType', function() {
+                return loadWithXhr({
+                    url : 'Data/htmlString.txt',
+                    responseType : 'json'
+                }).then(function(resolvedValue) {
+                    expect(resolvedValue).toBeUndefined();
+                }, function(rejectedError) {
+                    expect(rejectedError).toBeDefined();
+                    expect(rejectedError instanceof Error).toBe(true);
+                });
+            });
+        });
+
+        describe('returns a promise that resolves when the request loads', function() {
+            it('a non-null response with default responseType', function() {
+                return loadWithXhr({
+                    url : 'Data/Models/Box/ReadMe.txt'
+                }).then(function(resolvedValue) {
+                    expect(resolvedValue).toBe('CesiumBoxTest-NoTechnique.gltf is a modified glTF that has techniques, shaders & programs removed.');
+                }, function(rejectedError) {
+                    expect(rejectedError).toBeUndefined();
```

Comment:
There's no need to have a reject handler if you don't expect the promise to be rejected.  Cesium's scaffolding will automatically handle rejected promises and report an error.